src/transformers/modeling_tf_utils.py

```diff
@@ -1889,6 +1890,40 @@ def _get_resized_lm_head_bias(self, old_lm_head_bias, new_num_tokens):
 
         return new_lm_head_bias
 
+    def _v2_get_resized_lm_head_bias(
```

Comment:
The original `get_resized_lm_head_bias` actually adds weights to the model. Since those weights either have the same name (duplicated names -> save_weights fails) or a new name (i.e. can't map the new bias to the original bias at load time), model restoration with resized bias fails.

Also, the previous function was setting the bias with `trainable=True`, which is not the case for most models 😱 

The new function simply returns the new bias values, and delegates setting the bias to the model itself. For correct weight restoring capabilities, the bias weight has to be overwritten. Since different models have different bias layers, only the model itself can instantiate the layer -- see Bart's `set_bias()` changes.

Reply:
Wait, the bias on the LM head isn't trainable on most models? Is it initialized to match the frequency of tokens in the corpus and then frozen?

Reply:
Correct, depending on the model (e.g. [BERT's bias is trainable](https://github.com/huggingface/transformers/blob/70ba10e6d4684f56c50b740793498d4d2dd8a7bd/src/transformers/models/bert/modeling_tf_bert.py#L651)). When it is not trainable it can still be set, even if it stays as `0` the whole training procedure.

From the discussion on [slack](https://huggingface.slack.com/archives/C014N4749J9/p1661941064425229) regarding these untrainable bias: 
1. "the bias is saved as buffer [i.e. not trainable] because BART ties lm head with token embeddings, so the lm_head layer can’t have bias. Hence it’s added after applying  the lm_head."
2. "yeah and it's also explicitly a buffer to not compute gradients on it (...)"
3. "The bias is never updated at train time (...)"